app/controllers/blazer/queries_controller.rb

```diff
@@ -78,6 +78,14 @@ def show
     def edit
     end
 
+    def share
+      if params[:token] && params[:query_id] && params[:token] == Blazer.sharing.query_token(params[:query_id])
```

Comment:
This can leak the token via timing attacks, so compare in constant time:
```suggestion
      if params[:token] && params[:query_id] && ActiveSupport::SecurityUtils.secure_compare(params[:token], Blazer.sharing.query_token(params[:query_id]))
```